pkg/diagnostics/analyzers.go

```diff
@@ -150,6 +156,62 @@ func (a *analyzerFactory) eksaDockerAnalyzers() []*Analyze {
 	return append(analyazers, a.generateDeploymentAnalyzers(deployments)...)
 }
 
+// EksaLogTextAnalyzers given a slice of Collectors will check which namespaced log collectors are present
+// and return the log analyzers associated with the namespace in the namespaceLogTextAnalyzersMap
+func (a *analyzerFactory) EksaLogTextAnalyzers(collectors []*Collect) []*Analyze {
+	var analyzers []*Analyze
+	analyzersMap := a.namespaceLogTextAnalyzersMap()
+	for _, collector := range collectors {
+		if collector.Logs != nil {
+			analyzer, ok := analyzersMap[collector.Logs.Namespace]
+			if ok {
+				analyzers = append(analyzers, analyzer...)
+			}
+		}
+	}
+	return analyzers
+}
+
+// namespaceLogTextAnalyzersMap is used to associated log text analyzers with the logs collected from a specific namespace.
+// the key of the analyzers map is the namespace name, and the value are the associated log text analyzers.
+func (a *analyzerFactory) namespaceLogTextAnalyzersMap() map[string][]*Analyze {
```

Comment:
Seems like all the info coming from this method is static
curious why you decided to have methods instead of "static" vars

Reply:
my thought was it would be easier to add logic to the generation of the text analyzers in the future if necessary if they were struct methods